### PR TITLE
Prepare 0.72 release

### PR DESCRIPTION
**Pull request by ArthurSens**

### CODE REVIEW

**Review by dongjiang1989 — APPROVED** [review]
🎉LGTM

**Review by philipgough — APPROVED** [review]
lgtm

**Comment by slashpai** [comment]
want to get in https://github.com/prometheus-operator/prometheus-operator/pull/6333 also if possible :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+## 0.72.0 / 2024-02-23
+
+In this release we want to highlight the addition of ScrapeClasses.
+
+ScrapeClasses is an experimental feature that allows declaring multiple scrape configurations in the Prometheus/PrometheusAgent CRs that can be consumed by selected scrape-related resources, like ScrapeConfig, ServiceMonitor, PodMonitor and Probes. Initial support only allows the declaration of TLS configuration, but we plan to extend ScrapeClasses with more options in the future, e.g. Authorization, relabelings and scrape protocols.
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply CHANGELOG.md:5]: I'd use `objects` or `resources` instead of `CRs`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+
+* [FEATURE] Adds `ScrapeClasses` to Prometheus, PrometheusAgent, ScrapeConfig, ServiceMonitor, PodMonitor and Probe CRDs. #6199
```
Comment (simonpasquier) [inline CHANGELOG.md:7]:
```suggestion
* [FEATURE] Add `ScrapeClasses` to Prometheus, PrometheusAgent, ScrapeConfig, ServiceMonitor, PodMonitor and Probe CRDs. #6199
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+* [FEATURE] Add the `enableCompression` field to the ScrapeConfig CRD. #6236
+* [FEATURE] Add DigitalOcean service discovery to the ScrapeConfig CRD. #6220
+* [FEATURE] Add support for `scrapeProtocols` to the ScrapeConfig, ServiceMonitor and PodMonitor CRDs. #6235 #6268
+* [FEATURE] Add support for NS records in the DNS service discovery in the ScrapeConfig CRD. #6235
```
Comment (simonpasquier) [inline CHANGELOG.md:11]: (nit)
```suggestion
* [FEATURE] Add support for NS records to the DNS service discovery in the ScrapeConfig CRD. #6235
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+* [FEATURE] Add support for `keepFiringFor` in the ThanosRuler CRD. #6283
+* [ENHANCEMENT] Enable usage of sharding with ScrapeConfig. #6135
+* [BUGFIX] Correctly mount file subpaths in ThanosRuler CRD. #6243
+* [BUGFIX] Fix conversion of the `continue` field in AlertmanagerConfig between v1alpha1 and v1beta1 versions. #6282
+* [BUGFIX] Fixed failure of metadata informers when processing deleted objects. #6298
```
Comment (simonpasquier) [inline CHANGELOG.md:16]:
```suggestion
* [BUGFIX] Fix failure of metadata informers when processing deleted objects. #6298
```

**Review by simonpasquier — COMMENTED** [review]
just a few nits.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArthurSens) [reply CHANGELOG.md:5]: we declare on the custom resource, not in its definition 🤔 

Agreed that using CR can be confusing though, I'll change to something else

**Comment by slashpai** [comment]
https://github.com/prometheus-operator/prometheus-operator/pull/6333 is good for review now, if its ok to go in this release :)

**Comment by simonpasquier** [comment]
@ArthurSens fyi I've found a regression in #6199 which would need fixing before the release.

**Comment by ArthurSens** [comment]
> @ArthurSens fyi I've found a regression in #6199 which would need fixing before the release: https://github.com/prometheus-operator/prometheus-operator/pull/6199/files#r1504284344

Nice find, thanks for flagging it. I'll try to open a PR tomorrow
 
cc @nicolastakashi in case you can beat me to it :)

**Review by onedr0p — CHANGES_REQUESTED** [review]
It would be nice if this release also contained the update to the alertmanagerconfigs resource. A new field was added to the pushover_configs called `ttl`

Refs:
- https://github.com/prometheus/alertmanager/pull/3474
- https://prometheus.io/docs/alerting/latest/configuration/#pushover_config

**Review by buroa — CHANGES_REQUESTED** [review]
Would love to have the new TTL option in this release like @onedr0p said.

**Comment by ArthurSens** [comment]
> It would be nice if this release also contained the update to the alertmanagerconfigs resource. A new field was added to the pushover_configs called `ttl`
> 
> Refs:
> 
> * [Add missing ttl in pushhover prometheus/alertmanager#3474](https://github.com/prometheus/alertmanager/pull/3474)
> * https://prometheus.io/docs/alerting/latest/configuration/#pushover_config
> 
> Let me know if this is out of scope for this release and I'll open a separate issue instead.

First, we would need someone to implement this feature here in prometheus-operator. Since we don't have any PRs open implementing it, we can't add it to this release

**Pull request merged**